### ISSUE TITLE
Investigate adding markdownlint

### ISSUE THREAD

**Issue by XhmikosR** [twbs/bootstrap#29919]
Not sure how many FPs we'll get.

**Comment by mdo** [comment]
After fixing bullet inconsistencies...

```
README.md:196 MD045/no-alt-text Images should have alternate text (alt text)
README.md:197 MD045/no-alt-text Images should have alternate text (alt text)
README.md:198 MD045/no-alt-text Images should have alternate text (alt text)
README.md:199 MD045/no-alt-text Images should have alternate text (alt text)
README.md:200 MD045/no-alt-text Images should have alternate text (alt text)
README.md:201 MD045/no-alt-text Images should have alternate text (alt text)
README.md:202 MD045/no-alt-text Images should have alternate text (alt text)
README.md:203 MD045/no-alt-text Images should have alternate text (alt text)
README.md:204 MD045/no-alt-text Images should have alternate text (alt text)
README.md:205 MD045/no-alt-text Images should have alternate text (alt text)
site/content/docs/5.0/about/license.md:12:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
site/content/docs/5.0/components/button-group.md:22 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/components/button-group.md:22 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Ensure correct `role` and provide a label"]
site/content/docs/5.0/components/buttons.md:108 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/components/buttons.md:108 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Link functionality caveat"]
site/content/docs/5.0/components/dropdowns.md:805 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h4; Actual: h5]
site/content/docs/5.0/components/dropdowns.md:805 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### `data-toggle="dropdown"` still required"]
site/content/docs/5.0/components/navs.md:231 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "## Working with flex utilities"]
site/content/docs/5.0/components/popovers.md:102 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "#### Specific markup required for dismiss-on-next-click"]
site/content/docs/5.0/components/popovers.md:139 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "### Making popovers work for keyboard and assistive technology users"]
site/content/docs/5.0/components/popovers.md:294 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "#### Data attributes for individual popovers"]
site/content/docs/5.0/components/scrollspy.md:261 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "#### Resolvable ID targets required"]
site/content/docs/5.0/components/scrollspy.md:267 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "#### Non-visible target elements ignored"]
site/content/docs/5.0/components/tooltips.md:96 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/components/tooltips.md:96 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Overflow `auto` and `scroll`"]
site/content/docs/5.0/components/tooltips.md:113 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h4; Actual: h5]
site/content/docs/5.0/components/tooltips.md:113 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Making tooltips work for keyboard and assistive technology users"]
site/content/docs/5.0/components/tooltips.md:291 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "#### Data attributes for individual tooltips"]
site/content/docs/5.0/content/reboot.md:134 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "{{< /markdown >}}"]
site/content/docs/5.0/content/reboot.md:437 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/content/reboot.md:437 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### jQuery incompatibility"]
site/content/docs/5.0/content/tables.md:622 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/content/tables.md:622 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Vertical clipping/truncation"]
site/content/docs/5.0/customize/optimize.md:46:3 MD034/no-bare-urls Bare URL used [Context: "https://medium.com/dwarves-fou..."]
site/content/docs/5.0/customize/optimize.md:47:3 MD034/no-bare-urls Bare URL used [Context: "https://lukencode.com/2018/07/..."]
site/content/docs/5.0/extend/icons.md:29 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- [{{ .name }}]({{ .website }}..."]
site/content/docs/5.0/forms/layout.md:343 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "### Alternatives to hidden labels"]
site/content/docs/5.0/forms/overview.md:59 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/forms/overview.md:59 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Associating form text with form controls"]
site/content/docs/5.0/getting-started/introduction.md:121:37 MD034/no-bare-urls Bare URL used [Context: "https://twitter.com/{{<"]
site/content/docs/5.0/getting-started/introduction.md:128:48 MD034/no-bare-urls Bare URL used [Context: "https://twitter.com/{{<"]
site/content/docs/5.0/getting-started/javascript.md:29 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "## Incompatible plugins"]
site/content/docs/5.0/getting-started/javascript.md:49 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "## Selectors"]
site/content/docs/5.0/getting-started/javascript.md:72 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "## jQuery events"]
site/content/docs/5.0/getting-started/javascript.md:155 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h5]
site/content/docs/5.0/getting-started/javascript.md:155 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "##### Third-party libraries"]
site/content/docs/5.0/getting-started/rfs.md:11:32 MD034/no-bare-urls Bare URL used [Context: "https://github.com/twbs/rfs/tr..."]
site/content/docs/5.0/getting-started/rfs.md:17:132 MD034/no-bare-urls Bare URL used [Context: "https://github.com/twbs/rfs/tr..."]
site/content/docs/5.0/getting-started/rfs.md:86:138 MD034/no-bare-urls Bare URL used [Context: "https://github.com/twbs/rfs/tr..."]
site/content/docs/5.0/utilities/colors.md:47 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.bg-gradient-{{ .name }}`"]
site/content/docs/5.0/utilities/colors.md:53 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h4]
site/content/docs/5.0/utilities/colors.md:53 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "#### Dealing with specificity"]
site/content/docs/5.0/utilities/flex.md:26 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.d{{ .abbr }}-flex`"]
site/content/docs/5.0/utilities/flex.md:71 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.flex{{ .abbr }}-row`"]
site/content/docs/5.0/utilities/flex.md:124 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.justify-content{{ .abbr }}..."]
site/content/docs/5.0/utilities/flex.md:178 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.align-items{{ .abbr }}-sta..."]
site/content/docs/5.0/utilities/flex.md:232 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.align-self{{ .abbr }}-star..."]
site/content/docs/5.0/utilities/flex.md:258 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.flex{{ .abbr }}-fill`"]
site/content/docs/5.0/utilities/flex.md:289 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.flex{{ .abbr }}-{grow|shri..."]
site/content/docs/5.0/utilities/flex.md:415 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.flex{{ .abbr }}-nowrap`"]
site/content/docs/5.0/utilities/flex.md:440 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.order{{ $bp.abbr }}-{{ . }..."]
site/content/docs/5.0/utilities/flex.md:452 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.order{{ $bp.abbr }}-{{ . }..."]
site/content/docs/5.0/utilities/flex.md:615 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.align-content{{ .abbr }}-s..."]
site/content/docs/5.0/utilities/float.md:35 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `.float{{ .abbr }}-left`"]
```

**Comment by XhmikosR** [comment]
https://github.com/twbs/bootstrap/runs/1417773283:

```
 markdownlint-cli2 v0.0.11 (markdownlint v0.21.1)
Finding: **/*.md !**/node_modules/**
Linting: 102 file(s)
Summary: 17 error(s)
site/content/docs/5.0/components/toasts.md:160 MD046/code-block-style Code block style [Expected: fenced; Actual: indented]
site/content/docs/5.0/components/tooltips.md:116 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
site/content/docs/5.0/content/reboot.md:116:1 MD037/no-space-in-emphasis Spaces inside emphasis markers [Context: "* Lorem ipsum dolor sit amet"]
site/content/docs/5.0/content/reboot.md:118:1 MD037/no-space-in-emphasis Spaces inside emphasis markers [Context: "* Integer molestie lorem at ma..."]
site/content/docs/5.0/content/reboot.md:120:1 MD037/no-space-in-emphasis Spaces inside emphasis markers [Context: "* Nulla volutpat aliquam velit"]
site/content/docs/5.0/content/reboot.md:122:3 MD037/no-space-in-emphasis Spaces inside emphasis markers [Context: "* Purus sodales ultricies"]
site/content/docs/5.0/content/reboot.md:124:3 MD037/no-space-in-emphasis Spaces inside emphasis markers [Context: "* Ac tristique libero volutpat..."]
site/content/docs/5.0/content/reboot.md:126:1 MD037/no-space-in-emphasis Spaces inside emphasis markers [Context: "* Aenean sit amet erat nunc"]
site/content/docs/5.0/customize/optimize.md:51 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
site/content/docs/5.0/getting-started/introduction.md:154:37 MD034/no-bare-urls Bare URL used [Context: "https://twitter.com/"]
site/content/docs/5.0/getting-started/introduction.md:161:48 MD034/no-bare-urls Bare URL used [Context: "https://twitter.com/"]
site/content/docs/5.0/getting-started/javascript.md:80 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
site/content/docs/5.0/getting-started/rfs.md:11:32 MD034/no-bare-urls Bare URL used [Context: "https://github.com/twbs/rfs/tr..."]
site/content/docs/5.0/getting-started/rfs.md:17:132 MD034/no-bare-urls Bare URL used [Context: "https://github.com/twbs/rfs/tr..."]
site/content/docs/5.0/getting-started/rfs.md:86:138 MD034/no-bare-urls Bare URL used [Context: "https://github.com/twbs/rfs/tr..."]
site/content/docs/5.0/layout/columns.md:160 MD046/code-block-style Code block style [Expected: fenced; Actual: indented]
site/content/docs/5.0/layout/columns.md:177 MD046/code-block-style Code block style [Expected: fenced; Actual: indented]
```

Unfortunately, there's no way to ignore specific lines and most of these are false positives because we are using Hugo-specific markup.

Let's leave this open so that we track the issue.
